### ISSUE TITLE
Inside cloud fog doesn't reflect cloud's opacity

### ISSUE THREAD

**Issue by tacotexmex** [minetest/minetest#9060]
##### Minetest version
```
Minetest 5.0.1
```

##### OS / Hardware
Operating system: macOS
GPU model: Intel HD 4000
OpenGL version: 2.1

##### Summary
Regardless of cloud opacity its inside fog has the same thickness. It shouldn't be harder to see through a cloud from the outside compared to from within. Watch:

![screenshot_20191022_001304](https://user-images.githubusercontent.com/25328789/67247610-384c2f00-f462-11e9-8e44-be7826c88ded.png)
![screenshot_20191022_001307](https://user-images.githubusercontent.com/25328789/67247616-3c784c80-f462-11e9-807c-815f883dd0e9.png)

##### Steps to reproduce
1. Set semitransparent clouds.
2. Fly in and out out the cloud, watch surroundings.
<details>
  <summary>Here's a mod code snippet.</summary>
  <pre>minetest.register_on_joinplayer(function(player)
	player:set_clouds({
		density = 0.5,
		color = "#AABBCCAA",
		ambient = "#AABBCCAA",
		height = -1,
		thickness = 20,
		speed = 0
	})
end)
</pre>
</details>

**Comment by paramat** [comment]
Not really a bug, more a request for the 'inside cloud fog range' to reflect the alpha used in 'set clouds'. Clouds are usually chosen to be fairly opaque so this is usually not a problem.

**Comment by tacotexmex** [comment]
I think it’s fair to assume that players will assume that a very light opacity cloud doesn’t block the view just because you stand inside it.

Regardless of bug or feature, this blocks my gameplay mechanism from being developed so I’d appreciate any move on this. :)

**Comment by paramat** [comment]
Yes certainly fair.

This can certainly be improved, but consider that if a low alpha cloud increases fog distance (view range) to, for example, 64 nodes, any terrain further than 64 nodes away will still be completely obscured, which will still look very different to looking at a low alpha cloud from outside with terrain behind it. But also, view range in cloud has to be significantly lower than the normal player view range.

So, i suspect it will help customise view range fog when in a cloud, so that the fog starts very close to the player, so that there is haze even when the in-cloud view range is large.
The player view range will probably have to be taken into account to balance all this too.

**Comment by lhofhansl** [comment]
Yeah... I had pointed out the fact that the entire cloud when watched from the outside is less opaque than just part of it when standing inside - even with default settings.

Then again this was so that there *any* indication that you're inside a cloud. We can certainly tweak it... Perhaps just include cloudAlpha as a factor for the view-range inside the cloud, in fact we can probably just mutliply the view-range with it, and be done :).
In the extreme with alpha = 0 the clouds are not visible and so neither should the range change and with alpha = 1 we have completely opaque clouds and you should not see anything inside.

**Comment by paramat** [comment]
> In the extreme with alpha = 0 the clouds are not visible and so neither should the range change

I agree.
> with alpha = 1 we have completely opaque clouds and you should not see anything inside.

I disagree. In real life clouds appear opaque from the outside but inside it is possible to see something like 10+ metres, just like thick fog. So at alpha = 1 we can set view range to a sane minimum (16 nodes?).

I think 'fog start' should be set to 0 for all cloud alphas to make them more foggy inside.
